gradle-plugin/src/main/java/com/squareup/hephaestus/plugin/HephaestusPlugin.kt

```diff
@@ -22,26 +24,44 @@ open class HephaestusPlugin : Plugin<Project> {
   @ExperimentalStdlibApi
   override fun apply(project: Project) {
     val once = AtomicBoolean()
-    val innerApply = Action<AppliedPlugin> {
-      if (once.compareAndSet(false, true)) {
-        realApply(project)
+
+    fun PluginManager.withPluginOnce(
+      id: String,
+      action: (AppliedPlugin) -> Unit
+    ) {
+      withPlugin(id) {
+        if (once.compareAndSet(false, true)) {
+          action(it)
+        }
       }
     }
-    with(project.pluginManager) {
-      withPlugin("org.jetbrains.kotlin.android", innerApply)
-      withPlugin("org.jetbrains.kotlin.jvm", innerApply)
+
+    // Apply the Hephaestus plugin after the Kotlin plugin was applied. There could be a timing
+    // issue. Also make sure to apply it only once. A module could accidentally apply the JVM and
+    // Android Kotlin plugin.
```

Comment:
Could the ordering matter? eg: if I were to do

```
apply plugin: 'kotlin-android'
apply plugin: 'kotlin'
```

vs

```
apply plugin: 'kotlin'
apply plugin: 'kotlin-android'
```

that would have a difference in behaviour here, correct?

Reply:
I don't know who would want to do that. But at the same time I don't want to prevent that use case. Maybe in a future version the Kotlin plugin itself applies the plugins? I'd rather keep it as is.

Reply:
Sounds like you wanted to avoid that case, since you say above that applying both only happens "accidentally". I guess the behaviour here is undefined if that happens, depending on the ordering, hence why I brought it up.